Error running a new CA on linux/arm

### Subject of the issue
After going through the usual CA bootstrap process using step command, I'm not able to start step-ca using the default generated config file.

### Your environment
* OS - Raspbian GNU/Linux 10 (buster) - on Raspberry Pi 4 (4Gb RAM)
* Kernel Version  4.19.97-v7+ #1294 SMP

### Steps to reproduce
1. create a new CA using: `step ca init`
2. run the new CA using:  `step-ca /root/.step/config/ca.json --resolver=8.8.8.8:53`

### Expected behaviour
The new CA should be up & ready to serve connections.

### Actual behaviour
Error message on startup:
`Error opening database of Type badger with source /root/.step/db: error opening Badger database: Unable to mmap RDWR log file: cannot allocate memory`

### Additional context
I've tried rebuilding step-ca (after merging in master the changes from bagder2 branch), and the build fails:
`...\go\pkg\mod\github.com\dgraph-io\badger\v2@v2.0.1\value.go:859:38: constant 4294967295 overflows int`

used build command: 
`env GOOS=linux GOARCH=arm go build -ldflags="-s -w" -o step-ca`

--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,7 @@ all: build test lint
 
 bootstra%:
 	# Using a released version of golangci-lint to take into account custom replacements in their go.mod
-	$Q curl -sSfL https://raw.githubusercontent.com/smallstep/cli/master/make/golangci-install.sh | sh -s -- -b $(shell go env GOPATH)/bin v1.24.0
+	$Q curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.24.0
 
 .PHONY: bootstra%
 

--- a/Makefile
+++ b/Makefile
@@ -116,7 +116,7 @@ fmt:
 	$Q gofmt -l -w $(SRC)
 
 lint:
-	$Q LOG_LEVEL=error golangci-lint run
+	$Q LOG_LEVEL=error golangci-lint run --timeout=30m
 
 .PHONY: lint fmt
 

--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,7 @@ all: build test lint
 
 bootstra%:
 	# Using a released version of golangci-lint to take into account custom replacements in their go.mod
-	$Q curl -sSfL https://raw.githubusercontent.com/smallstep/cli/master/make/golangci-install.sh | sh -s -- -b $(shell go env GOPATH)/bin v1.23.8
+	$Q curl -sSfL https://raw.githubusercontent.com/smallstep/cli/master/make/golangci-install.sh | sh -s -- -b $(shell go env GOPATH)/bin v1.24.0
 
 .PHONY: bootstra%
 

--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,7 @@ all: build test lint
 
 bootstra%:
 	# Using a released version of golangci-lint to take into account custom replacements in their go.mod
-	$Q curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.24.0
+	$Q GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.24.0
 
 .PHONY: bootstra%
 

--- a/db/db.go
+++ b/db/db.go
@@ -35,10 +35,10 @@ type Config struct {
 	ValueDir   string `json:"valueDir,omitempty"`
 	Database   string `json:"database,omitempty"`
 
-	// ValueLogLoadingMode can be set to FileIO (instead of the default
-	// options.MemoryMap) to avoid memory-mapping log files. This can be useful
+	// BadgerFileLoadingMode can be set to 'FileIO' (instead of the default
+	// 'MemoryMap') to avoid memory-mapping log files. This can be useful
 	// in environments with low RAM
-	BadgerValueLogLoadingMode string `json:"badgerValueLogLoadingMode"`
+	BadgerFileLoadingMode string `json:"badgerFileLoadingMode"`
 }
 
 // AuthDB is an interface over an Authority DB client that implements a nosql.DB interface.
@@ -69,8 +69,8 @@ func New(c *Config) (AuthDB, error) {
 
 	opts := []nosql.Option{nosql.WithDatabase(c.Database),
 		nosql.WithValueDir(c.ValueDir)}
-	if len(c.BadgerValueLogLoadingMode) > 0 {
-		opts = append(opts, nosql.WithBadgerValueLogLoadingMode(c.BadgerValueLogLoadingMode))
+	if len(c.BadgerFileLoadingMode) > 0 {
+		opts = append(opts, nosql.WithBadgerFileLoadingMode(c.BadgerFileLoadingMode))
 	}
 
 	db, err := nosql.New(c.Type, c.DataSource, opts...)

--- a/db/db.go
+++ b/db/db.go
@@ -34,6 +34,11 @@ type Config struct {
 	DataSource string `json:"dataSource"`
 	ValueDir   string `json:"valueDir,omitempty"`
 	Database   string `json:"database,omitempty"`
+
+	// ValueLogLoadingMode can be set to FileIO (instead of the default
+	// options.MemoryMap) to avoid memory-mapping log files. This can be useful
+	// in environments with low RAM
+	BadgerValueLogLoadingMode string `json:"badgerValueLogLoadingMode"`
 }
 
 // AuthDB is an interface over an Authority DB client that implements a nosql.DB interface.
@@ -62,8 +67,13 @@ func New(c *Config) (AuthDB, error) {
 		return newSimpleDB(c)
 	}
 
-	db, err := nosql.New(c.Type, c.DataSource, nosql.WithDatabase(c.Database),
-		nosql.WithValueDir(c.ValueDir))
+	opts := []nosql.Option{nosql.WithDatabase(c.Database),
+		nosql.WithValueDir(c.ValueDir)}
+	if len(c.BadgerValueLogLoadingMode) > 0 {
+		opts = append(opts, nosql.WithBadgerValueLogLoadingMode(c.BadgerValueLogLoadingMode))
+	}
+
+	db, err := nosql.New(c.Type, c.DataSource, opts...)
 	if err != nil {
 		return nil, errors.Wrapf(err, "Error opening database of Type %s with source %s", c.Type, c.DataSource)
 	}

--- a/docs/database.md
+++ b/docs/database.md
@@ -42,12 +42,29 @@ more info).  Below are a few examples for supported databases:
   "key": ".step/secrets/intermediate_ca_key",
   "db": {
     "type": "badger",
-    "dataSource": "./stepdb",
-    "valueDir": "./steplogdb"   # leave empty if equivalent to dataSource
+    "dataSource": "./.step/db",
+    "valueDir": "./.step/valuedb"
+    "badgerValueLogLoadingMode": "MemoryMap"
   },
   ...
 },
 ```
+
+#### Options
+
+* `type`
+    * `badger` - currently refers to Badger V1. However, as Badger V1 is deprecated,
+    this will refer to Badger V2 starting with a the next major version release.
+    * `badgerV1` - explicitly select Badger V1.
+    * `badgerV2` - explicitly select Badger V2. Anyone looking to use Badger V2
+    will need to set it explicitly until it becomes the default.
+* `dataSource` - path, database directory.
+* `valueDir` [optional] - path, value directory, only if different from `dataSource`.
+* `badgerValueLogLoadingMode` [optional] - can be set to `FileIO` (instead of the default
+        `MemoryMap`) to avoid memory-mapping log files. This can be
+        useful in environments with low RAM.
+    * `MemoryMap` - default.
+    * `FileIO` - This can be useful in environments with low RAM.
 
 ### BoltDB
 

--- a/go.mod
+++ b/go.mod
@@ -13,7 +13,7 @@ require (
 	github.com/sirupsen/logrus v1.4.2
 	github.com/smallstep/assert v0.0.0-20200103212524-b99dc1097b15
 	github.com/smallstep/cli v0.14.2
-	github.com/smallstep/nosql v0.2.1-0.20200417162056-c1de767c9d0d
+	github.com/smallstep/nosql v0.2.1-0.20200421162603-b38671a21284
 	github.com/urfave/cli v1.22.2
 	golang.org/x/crypto v0.0.0-20200323165209-0ec3e9974c59
 	golang.org/x/net v0.0.0-20191209160850-c0dbc17a3553
@@ -24,4 +24,4 @@ require (
 )
 
 //replace github.com/smallstep/cli => ../cli
-replace github.com/smallstep/nosql => ../nosql
+//replace github.com/smallstep/nosql => ../nosql

--- a/go.mod
+++ b/go.mod
@@ -13,7 +13,7 @@ require (
 	github.com/sirupsen/logrus v1.4.2
 	github.com/smallstep/assert v0.0.0-20200103212524-b99dc1097b15
 	github.com/smallstep/cli v0.14.2
-	github.com/smallstep/nosql v0.2.0
+	github.com/smallstep/nosql v0.2.1-0.20200417162056-c1de767c9d0d
 	github.com/urfave/cli v1.22.2
 	golang.org/x/crypto v0.0.0-20200323165209-0ec3e9974c59
 	golang.org/x/net v0.0.0-20191209160850-c0dbc17a3553
@@ -24,3 +24,4 @@ require (
 )
 
 //replace github.com/smallstep/cli => ../cli
+replace github.com/smallstep/nosql => ../nosql

